### PR TITLE
[Merged by Bors] - feat(Algebra/Module/LinearMap/Basic): module structures of (semi)-linear maps over `DomMulAct` of a ring

### PR DESCRIPTION
**Pull request by jjaassoonn**
splitted from #8559

These are written by @alreadydone 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by alreadydone — APPROVED** [review]
Thanks!
maintainer merge

File: Mathlib/Algebra/Module/LinearMap/Basic.lean
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap/Basic.lean:1033]: This seems like it's in the wrong place; can't you put it near the other module instance on additive maps?

File: Mathlib/Algebra/Module/LinearMap/Basic.lean
Comment (jjaassoonn) [reply Mathlib/Algebra/Module/LinearMap/Basic.lean]: done

File: Mathlib/Algebra/Module/LinearMap/Basic.lean
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap/Basic.lean:856]: I renamed these lemmas to match the 6 or so that we already have. If you're not a fan of them, then let's rename all of them at once.

**Comment by eric-wieser** [comment]
bors d+

Thanks!

File: Mathlib/Algebra/Module/LinearMap/Basic.lean
Comment (jjaassoonn) [reply Mathlib/Algebra/Module/LinearMap/Basic.lean:856]: Thanks, I like your version better

**Comment by jjaassoonn** [comment]
Ones the CI passes, can you take another look before merging please @alreadydone

**Review by alreadydone — APPROVED** [review]
Thanks! LGTM

**Comment by jjaassoonn** [comment]
bors merge

**Pull request closed without merging**
